ui/src/components/pages/Pipeline/partials/Graph/partials/CustomEdge/index.tsx

```diff
@@ -64,34 +61,45 @@ const CustomEdge: FC<EdgeProps> = ({
 
   if (data?.selfEdge) {
     edgePath = `M ${sourceX} ${sourceY} L ${sourceX} ${
-      sourceY - 15
-    } Q ${sourceX} ${sourceY - 20} ${sourceX - 5} ${sourceY - 20} L ${
+      sourceY - 25
+    } Q ${sourceX} ${sourceY - 30} ${sourceX - 5} ${sourceY - 30} L ${
       targetX + 5
-    } ${targetY - 20} Q ${targetX} ${sourceY - 20} ${targetX} ${
-      sourceY - 15
-    } L ${targetX} ${sourceY}`;
+    } ${targetY - 30} Q ${targetX} ${sourceY - 30} ${targetX} ${
+      sourceY - 25
+    } L ${targetX} ${sourceY - 8}`;
     const centerX = (sourceX + targetX) / 2;
-    labelRenderer = `translate(-50%, -50%) translate(${centerX}px,${labelY}px)`;
+    labelRenderer = `translate(-50%, -50%) translate(${centerX}px,${
+      labelY - 10
+    }px)`;
   }
 
-  // Highlight the edge on isFull - thick red line
-  let edgeStyle, wmStyle, pendingStyle;
+  const { highlightValues, setHighlightValues } =
+    useContext<any>(HighlightContext);
 
-  if (data?.isFull) {
-    edgeStyle = {
-      stroke: "red",
-      strokeWidth: "0.125rem",
-      fontWeight: 700,
-    };
-  }
+  const color = data?.isFull ? "#DB334D" : "#8D9096";
+  useEffect(() => {
+    const handles = Array.from(
+      document.getElementsByClassName(
+        "react-flow__handle"
+      ) as HTMLCollectionOf<HTMLElement>
+    );
+    handles.forEach((handle) => {
+      handle.style.background = color;
+    });
+  }, [color]);
+
+  const edgeStyle = {
+    stroke: highlightValues[id] ? "black" : color,
+    strokeWidth: 2,
+  };
+  const wmStyle = {
+    color: color,
+  };
+  const pendingStyle = {
+    color: color,
+    fontWeight: data?.isFull ? 700 : 400,
+  };
 
-  if (sourceY === targetY || data?.backEdge) {
-    wmStyle = { marginTop: "-0.87rem" };
-    pendingStyle = { marginTop: "0.15rem" };
-  } else {
-    wmStyle = { right: "0.1rem" };
-    pendingStyle = { left: "0.1rem" };
-  }
   const getMinWM = () => {
     if (data?.edgeWatermark?.watermarks) {
       return Math.min.apply(null, data?.edgeWatermark?.watermarks);
```

Comment:
Was done prev to this PR, but "const getDelay = () => {" method should be changed to a useMemo, instead of calling it as method in every render.